spot-client/src/common/logger/post-to-endpoint.js

```diff
@@ -54,19 +96,83 @@ export default class PostToEndpoint {
             }
         }, []);
 
-        return fetch(
-            this._endpointUrl,
-            {
-                body: JSON.stringify({
-                    events: jsonEvents,
-                    'device_id': this._deviceId
-                }),
-                headers: {
-                    'Content-Type': 'application/json'
-                },
-                method: 'POST',
-                mode: 'cors'
-            }
-        );
+        this._requestsQueue.push(new PostLogsRequest(/* retry # */ 0, jsonEvents));
+
+        // Make sure that there's only 1 request on the flight by starting
+        // the reaction chain only if there's 1 request in the queue (the one
+        // added above).
+        if (this._requestsQueue.length === 1) {
+            this._sendLogs();
+        }
+    }
+
+    /**
+     * Tries to send first log request from the requests queue. In case of
+     * success it will keep taking requests of the queue until it's empty.
+     * On failure it will retry the request with a delay until
+     * the {@link PostToEndpoint#MAX_RETRIES} is reached.
+     *
+     * @returns {void}
+     * @private
+     */
+    _sendLogs() {
+        const request = this._requestsQueue.length && this._requestsQueue[0];
+
+        if (request) {
```

Comment:
Style comment: it might be easier to read if this were an early return because that's an exceptional/error case, plus it prevents having to remember the following code is within an if.